src/utils/typedData.ts

```diff
@@ -117,16 +129,66 @@ export function prepareSelector(selector: string): string {
  * Checks if the given Starknet type is a Merkle tree type.
  *
  * @param {StarknetType} type - The StarkNet type to check.
+ * @returns {boolean} True if the type is a Merkle tree type, false otherwise.
+ *
+ * @example
+ * ```typescript
+ * const result1;
+ * const merkleType = {
```

Comment:
@fishonamos we tested your examples  and they are wrong... we cannot merge chatGPT generated comments without properly testing